Content.Server/GameObjects/Components/Research/LatheComponent.cs

```diff
@@ -0,0 +1,163 @@
+using System.Collections.Generic;
+using Content.Server.GameObjects.Components.Stack;
+using Content.Server.GameObjects.EntitySystems;
+using Content.Shared.GameObjects.Components.Materials;
+using Content.Shared.GameObjects.Components.Research;
+using Content.Shared.Research;
+using Robust.Server.GameObjects;
+using Robust.Shared.GameObjects;
+using Robust.Shared.Interfaces.GameObjects;
+using Robust.Shared.Interfaces.GameObjects.Components;
+using Robust.Shared.Interfaces.Network;
+using Robust.Shared.Interfaces.Timers;
+using Robust.Shared.IoC;
+using Robust.Shared.Prototypes;
+using Robust.Shared.Timers;
+using Robust.Shared.Utility;
+using Robust.Shared.ViewVariables;
+
+namespace Content.Server.GameObjects.Components.Research
+{
+    public class LatheComponent : SharedLatheComponent, IAttackHand, IAttackBy, IActivate
+    {
+        public const int VolumePerSheet = 3750;
+
+        [Dependency]
+#pragma warning disable CS0649
+        private IPrototypeManager _prototypeManager;
+#pragma warning restore
+
+        [ViewVariables]
+        public Queue<LatheRecipePrototype> Queue => _queue;
+        private readonly Queue<LatheRecipePrototype> _queue = new Queue<LatheRecipePrototype>();
+
+        [ViewVariables]
+        public bool Producing => _producing;
+        private bool _producing = false;
+
+        private LatheRecipePrototype _producingRecipe = null;
+
+        internal bool Produce(LatheRecipePrototype recipe)
```

Comment:
I don't see any checks for whether this is blocked if the machine is already producing. Could lead to some bad behavior.

Reply:
Fixed.